airbyte-integrations/bases/base-normalization/normalization/transform_catalog/catalog_processor.py

```diff
@@ -0,0 +1,199 @@
+    targeted destination schema.
+
+    This is relying on a StreamProcessor to handle the conversion of a stream to a table one at a time.
+    """
+
+    def __init__(self, output_directory: str, integration_type: DestinationType):
+        """
+        @param output_directory is the path to the directory where this processor should write the resulting SQL files (DBT models)
+        @param integration_type is the destination type of warehouse
+        """
+        self.output_directory: str = output_directory
+        self.integration_type: DestinationType = integration_type
+        self.name_transformer: DestinationNameTransformer = DestinationNameTransformer(integration_type)
+
+    def process(self, catalog_file: str, json_column_name: str, target_schema: str):
+        """
+        This method first build the top-level streams and in a second loop will go over the substreams that
+        were nested. (breadth-first traversal)
+
+        @param catalog_file input AirbyteCatalog file in JSON Schema describing the structure of the raw data
+        @param json_column_name is the column name containing the JSON Blob with the raw data
+        @param target_schema is the final schema where to output the final transformed data to
+        """
+        # Registry of all tables in all schemas
+        tables_registry: Set[str] = set()
+        # Registry of source tables in each schemas
+        source_tables: Dict[str, Set[str]] = {}
+
+        catalog = read_json(catalog_file)
+        # print(json.dumps(catalog, separators=(",", ":")))
+        substreams = {}
+        for configured_stream in get_field(catalog, "streams", "Invalid Catalog: 'streams' is not defined in Catalog"):
+            stream_config = get_field(configured_stream, "stream", "Invalid Stream: 'stream' is not defined in Catalog streams")
+            schema_name = self.name_transformer.normalize_schema_name(target_schema)
+            raw_schema_name = self.name_transformer.normalize_schema_name(f"_airbyte_{target_schema}")
+            stream_name = get_field(stream_config, "name", f"Invalid Stream: 'name' is not defined in stream: {str(stream_config)}")
+            raw_table_name = self.name_transformer.normalize_table_name(f"_airbyte_raw_{stream_name}")
+
+            message = f"'json_schema'.'properties' are not defined for stream {stream_name}"
+            properties = get_field(get_field(stream_config, "json_schema", message), "properties", message)
+
+            from_table = "source('{}', '{}')".format(schema_name, raw_table_name)
+
+            # Check properties
+            if not properties:
+                raise EOFError("Invalid Catalog: Unexpected empty properties in catalog")
+
+            add_table_to_sources(source_tables, schema_name, raw_table_name)
+
+            stream_processor = StreamProcessor.create(
+                stream_name=stream_name,
+                output_directory=self.output_directory,
```

Comment:
would it be better to handle the generation of the model totally separately from persisting it to disk? this way it's easier to unit test stream processor